src/swaps/astroport/queries.js

```diff
@@ -0,0 +1,320 @@
+      msgs: [
+        new MsgExecuteContract(
+          address,
+          to,
+          {
+            swap: {
+              offer_asset: {
+                info: {
+                  native_token: {
+                    denom
+                  }
+                },
+                amount: quote.fromAmount
+              },
+              to: address
+            }
+          },
+          { [denom]: Number(quote.fromAmount) }
+        )
+      ],
+      gasLimit: 500_000
+    }
+  }
+}
+
+/*
+  This can handle swaps for:
+    1. ERC20 <-> ERC20
+    2. ERC20 <-> LUNA
+*/
+
+export const buildSwapFromContractTokenMsg = (
+  quote,
+  recipient,
+  fromTokenAddress,
+  toTokenAddress
+) => {
+  const isERC20ToLuna = quote.to === 'LUNA'
+  const isLunaToERC20 = quote.from === 'LUNA'
+  const isERC20ToERC20 = quote.to !== 'LUNA' && quote.from !== 'LUNA'
+
+  const swapMsg = {
+    execute_swap_operations: {
+      offer_amount: quote.fromAmount,
+      operations: [
+        {
+          astro_swap: {
+            ...((isERC20ToERC20 || isERC20ToLuna) && {
+              offer_asset_info: {
+                token: {
+                  contract_addr: fromTokenAddress
+                }
+              }
+            }),
+            ...(isLunaToERC20 && {
+              offer_asset_info: {
+                native_token: {
+                  denom: 'uluna'
+                }
+              }
+            }),
+            ask_asset_info: {
+              native_token: {
+                denom: 'uusd'
+              }
+            }
+          }
+        },
+        {
+          astro_swap: {
+            offer_asset_info: {
+              native_token: {
+                denom: 'uusd'
+              }
+            },
+            ...((isLunaToERC20 || isERC20ToERC20) && {
+              ask_asset_info: {
+                token: {
+                  contract_addr: toTokenAddress
+                }
+              }
+            }),
+            ...(isERC20ToLuna && {
+              ask_asset_info: {
+                native_token: {
+                  denom: 'uluna'
+                }
+              }
+            })
+          }
+        }
+      ]
+    }
+  }
+
+  if (isERC20ToERC20 || isERC20ToLuna) {
+    const msgInBase64 = Buffer.from(JSON.stringify(swapMsg)).toString('base64')
+
+    return {
+      data: {
+        fee: quote.fee,
+        msgs: [
+          new MsgExecuteContract(recipient, fromTokenAddress, {
+            send: {
+              msg: msgInBase64,
+              amount: quote.fromAmount,
+              contract: 'terra16t7dpwwgx9n3lq6l6te3753lsjqwhxwpday9zx' // USE FOR ERC20 <-> ERC20 AND ERC20 <-> LUNA Swaps
+            }
+          })
+        ],
+        gasLimit: 1_500_000
+      }
+    }
+  }
+
+  return {
+    data: {
+      fee: quote.fee,
+      msgs: [
+        new MsgExecuteContract(
+          recipient,
+          'terra16t7dpwwgx9n3lq6l6te3753lsjqwhxwpday9zx', // USE for Luna <-> ERC20 Swaps
```

Comment:
Make this address and addresses like this constant?

Reply:
done :)